common/prque/sstack.go

```diff
@@ -94,7 +96,11 @@ func (s *sstack) Len() int {
 // Compares the priority of two elements of the stack (higher is first).
 // Required by sort.Interface.
 func (s *sstack) Less(i, j int) bool {
-	return (s.blocks[i/blockSize][i%blockSize].priority - s.blocks[j/blockSize][j%blockSize].priority) > 0
+	a, b := s.blocks[i/blockSize][i%blockSize].priority, s.blocks[j/blockSize][j%blockSize].priority
+	if s.wrapAround {
```

Comment:
What's the difference between the `a>b` and `a-b>0`?

Reply:
The way they handle numeric overflows. For example, `MaxInt64 > MinInt64` but `MaxInt64-MinInt64 < 0`. If we use the latter type of comparison then the individual priority values are allowed to overflow and "wrap around" the 64 bit range many times as long as the biggest difference between two priorities in the queue at any moment can be expressed in the `int64` range. `flowcontrol.ClientManager` is a good example for this mode where there is a cumulative "buffer recharge integrator" that is continuously being increased and does overflow. `rcQueue` item priorities are all calculated relative to this value and the difference between any two item priorities fits safely into `int64`. The first (free clients only) client pool design also did take advantage of this feature but the current priority pool scheme uses absolute values.
This "wrap around" feature requires care to use and until now all `prque.Prque`-s operated in this mode. Now it's only used if the queue is explicitly created with the `NewWrapAround` constructor.

Reply:
This PR reverts to the non-tricky version in most of the cases and does not change the behavior of the client manager. I will change client manager's priority calculation and get rid of thip wrap-around trick altogether in a next PR.

Reply:
Thanks!